### DISCUSSION TITLE
Inconsistency between JavaVirtualMachine.getEstimatedJVMStartTime() and JavaVirtualMachine.getJVMTerminationTime() when log file does not contain timestamps?

### DISCUSSION THREAD

**Discussion by Bluetopia** [microsoft/gctoolkit#338] [Q&A] [answered]
Using the attached sample log file, I've run into an issue where I've been unable to compare the start/end DateTimeStamps.

[sample.zip](https://github.com/microsoft/gctoolkit/files/14572842/sample.zip)

After loading the file, jvm.getEstimatedJVMStartTime() returns a ZonedDateTime populated DateTimeStamp (Timestamp = NaN).  However, jvm.getJVMTerminationTime() returns a timestamp populated DateTimeStamp (dateTime = null).  Also, the timestamp provided is 1.193, which is not correct as this file spans 5 days.

Using debugging, I see that GCToolkit has populated a UnifiedJavaVirtualMachine for this.  I did notice there was a previous NPE issue around Unified logging and logs missing timestamps: https://github.com/microsoft/gctoolkit/issues/274, though it's not clear if the problem was addressed. (A commit is associated with the issue, but the issue remains open)

Should GCToolkit be smart enough to use datestamps for calculating start/end dates if a JVM Epoch timestamp isn't present?

**Comment by karianna** [comment, marked as answer]
@Bluetopia - If you build tip (3.0.5-SNAPSHOT) I think this should be fixed, can you give that a go?

**Reply by Bluetopia** [reply]
Yep, the problem is resolved with the current headrev.  Hopefully 3.0.5 will be coming soon :)

Thanks!

**Reply by Bluetopia** [reply]
@karianna Spoke too soon, sort of. 

So, the new build works with UnifiedJavaVirtualMachine, but in testing some Pre-unified data, I'm getting an NPE. 

Is it worth unmarking the answer and continuing in this thread, or should I post a new bug or discussion with the call stack since it's a similar issue in coming from a different part of the library?  (Same method impacted, but data was likely generated differently as it is a pre-unified log with timestamps only)

**Reply by karianna** [reply]
@Bluetopia A new bug with the call stack would be great thank you!
